### DISCUSSION TITLE
Possible Confusing Reflected Light Sensor Readings Based Due To Ambient Light

### DISCUSSION THREAD

**Discussion by sofalights** [pybricks/support#1921] [General]
I have no idea whether this is because of my spike prime colour sensor having issues, but ambient light possibly affects the reflected light reading,
HOWEVER, in a confusing way.

My current setup is as follows:
A yellowish light source is placed approximately 165 cm high from a white plastic board.
2 sensors are 0.8 cm perpendicular to the white plastic board, set to read reflected hsv. (they are about 1.6cm apart)

The results are:
```
180 0 100  left sens
180 0 100  right sens
```

However, when tested under shade (under the table) the results are:
```
0 0 100  left sens
0 0 100  right sens
```

To check again, here are 2 more readings under the first conditions:
```
180 7 100  left sens
180 0 99  right sens
```
```
180 0 100  left sens
174 7 100  right sens
```
Due to these readings on the same conditions, I am unable to ascertain whether these are faulty colour sensors.
However, due to the fact that the hue readings correlate to cyan, but the ambient light is yellow, is a matter which should be brought to attention.

**Comment by sofalights** [comment]
First conditions (sorry for broken camera, btw you can’t see the yellow lighting in the picture)

![D9E20FF7-8F61-4BE6-B5E7-FDE28BE935FA](https://github.com/user-attachments/assets/f1927a6d-aa30-460e-869e-f4494de53ae1)

Second conditions (under table)

![28B56FF0-9205-432D-A0AF-4670719FB9B3](https://github.com/user-attachments/assets/6c54dfad-6b5c-4a3b-a815-5b7fe04f80b9)

**Reply by sofalights** [reply]
I am bringing this matter up as this may affect people using hsv for line-following robots.

**Comment by laurensvalk** [comment]
This is the correct behavior.

On low saturation surfaces (s is low) like a white table, the hue value doesn't mean anything (it is the color shade, but only if there is any color to see). So it may appear to jump around, but that is not something to worry about.

This is why for color detection we usually recommend working with the detectable colors setting. You can find examples of this in the documentation. Then you don't have to build your own color matching mechanism.

**Reply by sofalights** [reply]
Ah, I see. Thank you! I'll close the discussion as outdated then
